worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/adapter/BukkitImplAdapter.java

```diff
@@ -178,6 +182,13 @@ default boolean simulateItemUse(World world, BlockVector3 position, BaseItem ite
      */
     BaseItemStack adapt(ItemStack itemStack);
 
+    /**
+     * Get the {@link SideEffect}s that this adapter supports.
+     *
+     * @return The side effects that are supported
+     */
+    Collection<SideEffect> getSupportedSideEffects();
```

Comment:
I would prefer if we enforced this to be a `Set`. I don't think having multiple of the same `SideEffect` has any meaning.